crates/turborepo-auth/src/server/login_server.rs

```diff
@@ -0,0 +1,59 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::Result;
+use async_trait::async_trait;
+use axum::{extract::Query, response::Redirect, routing::get, Router};
+use serde::Deserialize;
+use tokio::sync::OnceCell;
+
+#[derive(Debug, Clone, Deserialize)]
+struct LoginPayload {
+    token: String,
+}
+
+#[async_trait]
+pub trait LoginServer {
+    async fn run(
+        &self,
+        port: u16,
+        login_url_base: String,
+        login_token: Arc<OnceCell<String>>,
+    ) -> Result<()>;
+}
+
+/// TODO: Document this.
+pub struct DefaultLoginServer;
+impl DefaultLoginServer {
```

Comment:
You can derive `Default` on this too. Although tbh you could just constrict it literally since that's even shorter.

Reply:
Haven't seen / worked with `derive(Default)` yet, I'll look into that